libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -961,8 +961,31 @@ void CompilerUtils::convertType(
 	}
 	case Type::Category::Array:
 	{
-		solAssert(targetTypeCategory == stackTypeCategory, "");
 		auto const& typeOnStack = dynamic_cast<ArrayType const&>(_typeOnStack);
+		if (_targetType.category() == Type::Category::FixedBytes)
+		{
+			solAssert(
+				typeOnStack.isByteArray() && !typeOnStack.isString(),
+				"Array types other than bytes not convertible to bytesNN."
+			);
+			solAssert(typeOnStack.isDynamicallySized(), "");
```

Comment:
These three settings suggest to me we may want to have something like `isDynamicBytes()` at some point?

Reply:
Definitely, we need to check if it occurs in multiple places and add such helper